### PR TITLE
Make `GaussianNormalizer` distribution names consistent with `scipy`

### LINKED ISSUE #656: The `GaussianNormalizer` should accept distribution names that are consistent with `scipy`
### Problem Description
The [GaussianNormalizer](https://docs.sdv.dev/rdt/transformers-glossary/numerical/gaussiannormalizer) accepts a distribution parameter, where you can provide a string name of the univariate distribution. 

The acceptable strings are: `'gaussian'`, `'gamma'`, `'beta'`, `'student_t'`, `'truncated_gaussian'` and `'gaussian_kde'`

These names are **inconsistent** with both of the following:
- The underlying scipy library where the univariates come from and
- The [SDV library](https://docs.sdv.dev/sdv/single-table-data/modeling/synthesizers/gaussiancopulasynthesizer) which uses the same names as scipy.

### Expected behavior
Change the mappings to the following:

- `'norm'` should be used for a Gaussian distribution
- `'truncnorm'` should be used for a truncated Gaussian Distribution
- `'t'` should be used for Student T

### Additional context
In order to keep backwards compatibility, we can still accept the older names. If a user inputs them, internally use the newer names and show a Future Warning.

```python
# example: user inputs 'gaussian' instead of 'norm'
my_transformer = GaussianNormalizer(distribution='gaussian')
```
This ultimately uses the correct distribution but it shows a future warning.
```
FutureWarning: Future versions of RDT will not support 'gaussian' as an option. Please use 'norm' instead.
```

### PR DESCRIPTION
**Pull request by fealho**
Resolve #656.

### CODE REVIEW

File: rdt/transformers/numerical.py [hunk 1]
```diff
@@ -311,6 +315,15 @@ def __init__(self, model_missing_values=None, learn_rounding_scheme=False,
 
         self._distributions = self._get_distributions()
         if isinstance(distribution, str):
+            if distribution in {'gaussian', 'student_t', 'truncated_gaussian'}:
+                deprecated_distributions_mapping = {
```
Comment (rwedge) [inline rdt/transformers/numerical.py:319]: could we treat this mapping as a private class constant?

File: rdt/transformers/numerical.py [hunk 2]
```diff
@@ -285,13 +285,17 @@ def _get_distributions():
...
-            'student_t': univariate.StudentTUnivariate,
+            't': univariate.StudentTUnivariate,
             'gaussian_kde': univariate.GaussianKDE,
-            'truncated_gaussian': univariate.TruncatedGaussian,
+            'truncnorm': univariate.TruncatedGaussian,
             'uniform': univariate.UniformUnivariate,
+            # the following are deprecated
+            'gaussian': univariate.GaussianUnivariate,
+            'student_t': univariate.StudentTUnivariate,
+            'truncated_gaussian': univariate.TruncatedGaussian,
```
Comment (rwedge) [inline rdt/transformers/numerical.py:298]: if in the init we replaced the deprecated distribution string with the new one before calling this method we could remove the deprecated mappings from this dictionary

Comment (fealho) [reply rdt/transformers/numerical.py]: That's a good suggestion, thanks.

**Review by rwedge — APPROVED** [review]
Looks good, one more suggestion

**Review by pvk-developer — APPROVED** [review]

**Pull request merged**